toolchain/check/handle_function.cpp

```diff
@@ -96,16 +109,16 @@ static auto BuildFunctionDecl(Context& context,
       .PopAndDiscardSoloNodeId<Parse::NodeKind::FunctionIntroducer>();
 
   // Process modifiers.
-  auto modifiers = DiagnoseModifiers(context, name_context.target_scope_id);
+  auto modifiers =
+      DiagnoseModifiers(context, is_definition, name_context.target_scope_id);
   if (!!(modifiers & KeywordModifierSet::Access)) {
     context.TODO(context.decl_state_stack().innermost().modifier_node_id(
                      ModifierOrder::Access),
                  "access modifier");
   }
-  if (!!(modifiers & KeywordModifierSet::Extern)) {
-    context.TODO(context.decl_state_stack().innermost().modifier_node_id(
-                     ModifierOrder::Extern),
-                 "extern modifier");
+  bool is_extern = !!(modifiers & KeywordModifierSet::Extern);
+  if (is_extern && is_definition) {
+    is_extern = false;
```

Comment:
It would seem nice if `DiagnoseModifiers` pruned modifiers that it diagnosed for you. Then you could simplify this.

Maybe a TODO? Doesn't seem worth blocking on, just maybe a nice refactoring to queue up.